### PR TITLE
Fix extrapolation in ZNE function

### PR DESCRIPTION
**Pull request by dime10**
Extrapolation was done against the folding numbers instead of the scale factors. Since the folding numbers start at 0, extrapolation would always yield a result very close to the first data point.

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -273,6 +273,10 @@
 
 <h3>Bug fixes</h3>
 
+* Fix a bug in the extrapolation part of the `catalyst.mitigate_with_zne` function that would lead
```
Comment (cosenal) [inline doc/releases/changelog-dev.md:276]: nit: the sentence reads a bit redundant. A bug in extrapolation code leads to..incorrect extrapolated result"

File: frontend/catalyst/api_extensions/error_mitigation.py [hunk 2]
```diff
@@ -164,14 +162,14 @@ class ZNECallable(CatalystCallable):
     def __init__(
         self,
         fn: Callable,
-        num_folds: jnp.ndarray,
+        scale_factors: Sequence,
```
Comment (cosenal) [inline frontend/catalyst/api_extensions/error_mitigation.py:165]: can type them more strictly as `[int]`

File: frontend/catalyst/api_extensions/error_mitigation.py [hunk 3]
```diff
@@ -209,16 +207,18 @@ def __call__(self, *args, **kwargs):
             callable_fn
         ), "expected callable set as param on the first operation in zne target"
 
-        results = zne_p.bind(
-            *args_data, self.num_folds, folding=folding, jaxpr=jaxpr, fn=callable_fn
+        fold_numbers = (jnp.asarray(self.scale_factors, dtype=int) - 1) // 2
```
Comment (cosenal) [inline frontend/catalyst/api_extensions/error_mitigation.py:210]: nice that they became element-wise ops 🤙🏻

**Review by cosenal — APPROVED** [review]

File: frontend/catalyst/api_extensions/error_mitigation.py (hunk 2 above)
Comment (dime10) [reply frontend/catalyst/api_extensions/error_mitigation.py:165]:
```suggestion
        scale_factors: Sequence[int],
```

File: doc/releases/changelog-dev.md (hunk 1 above)
Comment (dime10) [reply doc/releases/changelog-dev.md:276]:
```suggestion
* Fix a bug in `catalyst.mitigate_with_zne` that would lead
```

**Review by rmoyard — APPROVED** [review]
Good catch, I have fixed a test and it looks good to be merged.

**Pull request merged**